sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -2541,6 +2598,39 @@ protected internal virtual QueueClient WithClientSideEncryptionOptionsCore(Clien
                 }
             }
         }
+
+        private async Task<PeekedMessage[]> ToPeekedMessagesWithInvalidMessageHandling(
+            IEnumerable<PeekedMessageItem> peekedMessageItems,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            List<PeekedMessage> peekedMessages = new List<PeekedMessage>();
+
+            foreach (var peekedMessageItem in peekedMessageItems)
+            {
+                try
+                {
+                    peekedMessages.Add(PeekedMessage.ToPeekedMessage(peekedMessageItem, ClientConfiguration.MessageEncoding));
+                }
+                catch (FormatException)
```

Comment:
Same comment about moving the `ClientConfiguration.InvalidMessageHandler != null` logic here.